### DISCUSSION TITLE
More information for teardown in LightningDataModule

### DISCUSSION THREAD

**Discussion by yipliu** [ashleve/lightning-hydra-template#451] [Q&A] [answered]
Hi,

I found that a new function [teardown](https://github.com/ashleve/lightning-hydra-template/blob/89f9f02fd41c0dd1c1deaa338c796a7a7cb26bc0/src/datamodules/mnist_datamodule.py#L117) has been added in `MNISTDataModule`, but all the descriptions of it on the web say to clean up the data.

Does it clean up the data on the memory? If so, it is very useful when we are working with very large data sets.

How do we use this function?

```python
def teardown(self, stage: Optional[str] = None):
    """Clean up after fit or test."""
    pass
```

Don't write anything?

**Comment by ashleve** [comment, marked as answer]
According to lightning docs:
https://pytorch-lightning.readthedocs.io/en/latest/data/datamodule.html#teardown

The teardown is just a handy function called at the end of fit, validate, test, or predict. 

It's up to you to decide what the function should be doing.

It is called from every process across all the nodes.

The`stage` arg informs us whether the current stage is "fit", "validate", etc.

**Comment by yipliu** [comment]
Get it.

In general, what do we do with this function？

I hardly see any examples of using this function.

**Reply by ashleve** [reply]
- cleanning up downloaded data after training finishes
- closing any open connections a dataloader makes
- etc.

**Comment by nodiper** [comment]
def teardown(self, stage: Optional[str] = None):
    """Clean up after fit or test."""
    pass
